### PR TITLE
implement `with_capacity` for `WriteBatch`

### PR DESCRIPTION
**Pull request by 0xdeafbeef**

### CODE REVIEW

File: clippy.toml
Comment (zaidoon1) [inline clippy.toml:1]: why was this deleted?

Comment (zaidoon1) [reply clippy.toml:1]: oh i see, we can keep Cargo.toml then

File: src/write_batch.rs [hunk 1]
```diff
@@ -89,6 +89,22 @@ unsafe extern "C" fn writebatch_delete_callback(state: *mut c_void, k: *const c_
...
 impl<const TRANSACTION: bool> WriteBatchWithTransaction<TRANSACTION> {
+    /// Create a new `WriteBatch` without allocating memory.
+    pub fn new() -> Self {
+        Self {
+            inner: unsafe { ffi::rocksdb_writebatch_create() },
+        }
+    }
+
+    /// Creates `WriteBatch` with the specified `capacity` in bytes. Allocates immediately.
+    pub fn with_capacity(capacity: usize) -> Self {
```
Comment (zaidoon1) [inline src/write_batch.rs:100]: nit: might be safer to name this `with_capacity_bytes` or something like that. Typically with_capacity in vecs, etc.. is for number of elements and I can see somehow not reading the api docs and just using the number of kvs they want to put into the db as the value which would be bad.. But not sure..

Comment (0xdeafbeef) [reply src/write_batch.rs]: I've renamed it to `with_capacity_bytes`

**Review by zaidoon1 — APPROVED** [review]

**Review by stanislav-tkach — APPROVED** [review]
